### PR TITLE
Create package structure for YOLO implementation

### LINKED ISSUE #92: Update ultralytics_ros
The current state of the _ultralytics_ros_ package in our system is not up-to-date in the main repository due to changes made locally on the vehicle. This mismatch creates confusion and complicates maintaining consistency. To address this issue and integrate the vision component into our main repository, I propose the following package structure:

```
auv-software/
  -auv_vision/
     -launch
     -models (empty)
     -CMakeLists.txt
     -package.xml
```

With this structure, we can include the vision module in our main repository while preventing inconsistencies caused by local modifications. The launch files in the package will handle configuring the required model and input/output topics via arguments, making further modifications to the _ultralytics_ros_ package unnecessary. Having the package installed will suffice.

Additionally, I believe it would be more appropriate to share our AI models securely with the software team through a private platform. In the future, we might also consider merging the _auv_cam_ package into the _auv_vision_ package.

What are your thoughts on this, @senceryazici ?

### PR DESCRIPTION
**Pull request by ozanhakantunca**
Closes #92 

The auv_vision meta-package and the auv_detection package have been created and properly configured. With this package structure, we can include our custom competition model without modifying third party repos and start the detection process.

### CODE REVIEW

File: auv_vision/auv_detection/launch/tracker.launch [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (senceryazici) [inline auv_vision/auv_detection/launch/tracker.launch:1]: start.launch?

File: auv_vision/auv_vision/package.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <version>0.1.0</version>
+  <description>The auv_vision package</description>
+
+  <maintainer email="mimarlarf@gmail.com">Faruk Mimarlar</maintainer>
+  <author email="mimarlarf@gmail.com">Faruk Mimarlar</author>
+
+  <license>BSD-3-Clause</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <depend>auv_detection</depend>
```
Comment (senceryazici) [inline auv_vision/auv_vision/package.xml:13]: you must add

```xml

<depend>auv_vision</depend>

```

to `auv_software`'s `package.xml`

File: auv_vision/auv_detection/package.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  <version>0.1.0</version>
+  <description>The auv_detection package</description>
+
+  <author email="t.ozanhakan@gmail.com">Ozan Hakan Tunca</author>
+  <maintainer email="t.ozanhakan@gmail.com">Ozan Hakan Tunca</maintainer>
+
+  <license>BSD-3-Clause</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
```
Comment (senceryazici) [inline auv_vision/auv_detection/package.xml:13]: This package launches `ultralytics_ros` right? where is the dependency?

File: auv_vision/auv_detection/launch/tracker.launch [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    <param name="result_line_width" value="$(arg result_line_width)"/>
+    <param name="result_font_size" value="$(arg result_font_size)"/>
+    <param name="result_font" value="$(arg result_font)"/>
+    <param name="result_labels" value="$(arg result_labels)"/>
+    <param name="result_boxes" value="$(arg result_boxes)"/>
+    <rosparam param="classes" subst_value="true">$(arg classes)</rosparam>
+    <rosparam param="device" subst_value="true">$(arg device)</rosparam>
+  </node>
+
+  <node if="$(arg debug)" name="image_view" pkg="image_view" type="image_view" output="screen">
```
Comment (senceryazici) [inline auv_vision/auv_detection/launch/tracker.launch:44]: we usually use `use_gui` argument for applications that run with GUI. can you replace this?

File: auv_vision/auv_detection/package.xml (hunk 3 above)
Comment (senceryazici) [reply auv_vision/auv_detection/package.xml:13]: Also, it seems

`image_view` is being used too.

**Review by senceryazici — DISMISSED** [review]
requested small changes

**Comment by ozanhakantunca** [comment]
Added requested changes.

**Review by senceryazici — DISMISSED** [review]

**Pull request merged**